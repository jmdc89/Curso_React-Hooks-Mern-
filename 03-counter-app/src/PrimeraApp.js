import React from 'react';
import PropTypes from 'prop-types';

// FC
const PrimeraApp = ( { saludo, subtitulo} ) => {

    return (
        <>
        <h1> { saludo }</h1>
        {/* {<pre> { JSON.stringify( saludo, null, 3) }</pre>} */}
        <p>Mi primera aplicación</p>
        </>
    );
}

//De esta forma se obliga a enviar al componente las propiedades necesarias
// PrimeraApp.PropTypes = {
//     saludo: PropTypes.string.isRequired
// }

PrimeraApp.defaultProps = {
    saludo: 'Hola Mundo',
    subtitulo: 'Soy un subtítulo'
}


export default PrimeraApp;