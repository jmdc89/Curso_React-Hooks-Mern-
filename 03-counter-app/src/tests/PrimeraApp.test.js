
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('Pruebas en PrimeraApp', () => {

    // test('debe de mostrar el mensaje "Hola,Soy Goku"', () => {

    //     const saludo = 'Hola, soy Goku';

    //     const { getByText } = render( <PrimeraApp saludo={ saludo } />> );

    //     expect( getByText(saludo)).toBeInTheDocument();
    // });

    describe('Pruebas en <PrimeraApp />', () => {

        test('debe de mostrar <PrimeraApp /> correctamente', () => {

            const saludo = 'Hola, Soy Goku';
            const wrapper = shallow( <PrimeraApp saludo={ saludo } /> )

            expect( wrapper ).toMatchSnapshot();

        });
    });


});