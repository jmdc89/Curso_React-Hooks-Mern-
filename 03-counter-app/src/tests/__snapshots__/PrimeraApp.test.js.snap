// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, Soy Goku
  </h1>
  <p>
    Mi primera aplicaci√≥n
  </p>
</Fragment>
`;
